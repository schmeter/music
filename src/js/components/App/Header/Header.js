import React from 'react';
import PropTypes from 'prop-types';
import autobind from 'autobind-decorator';

import Link from '../../Link';
import Icon from '../../Icon';
import i18n from '../../../services/i18n';
import { getUrl } from '../../../services/navigation';

class Header extends React.Component {
    @autobind
    handleClickLink() {
        const { openLayer } = this.props;

        openLayer('info');
    }

    render() {
        return (
            <header className="main-header">
                <h1 className="main-headline">
                    <Link to={getUrl('index')}>
                        {i18n(`app_title`)}
                    </Link>
                </h1>
                <Link
                    className="button header-button home-button"
                    to={getUrl('index')}
                >
                    <Icon id="chevron-circle-left" />
                </Link>
                <button
                    className="button header-button info-button"
                    onClick={this.handleClickLink}
                    tabIndex="-1"
                >
                    <Icon id="info-circle" />
                </button>
            </header>
        );
    }
}

Header.propTypes = {
    openLayer: PropTypes.func.isRequired,
};

export default Header;
