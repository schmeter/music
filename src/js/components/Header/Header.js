import React, { useCallback } from 'react';
import PropTypes from 'prop-types';

import Link from '../Link';
import Icon from '../Icon';
import i18n from '../../helpers/i18n';
import { getRouteWithParams } from '../../helpers/navigation';

const Header = ({ openLayer }) => {
  const handleClickInfo = useCallback(() => {
    openLayer('info');
  }, [
    openLayer,
  ]);

  const handleClickMenu = useCallback(() => {
    openLayer('menu');
  }, [
    openLayer,
  ]);

  return (
    <header className="main-header">
      <h1 className="main-headline">
        <Link to={getRouteWithParams('index')}>
          {i18n('app_title')}
        </Link>
      </h1>
      <button
        className="button header-button menu-button left"
        onClick={handleClickMenu}
      >
        <Icon id="bars" />
      </button>
      <button
        className="button header-button info-button right"
        onClick={handleClickInfo}
      >
        <Icon id="heartbeat" />
      </button>
    </header>
  );
};

Header.propTypes = {
  openLayer: PropTypes.func.isRequired,
};

export default Header;
