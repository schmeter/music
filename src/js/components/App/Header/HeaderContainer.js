import { connect } from 'react-redux';

import Header from './Header';

import { openLayerAction } from '../../../state/actions/layer';


export const mapDispatchToProps = (dispatch) => ({
    openLayer: (id) => dispatch(openLayerAction(id))
});

export default connect(null, mapDispatchToProps)(Header);
