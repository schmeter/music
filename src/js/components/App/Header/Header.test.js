import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from './Header';

const mockedEvent = {
    preventDefault: jest.fn()
};

it('renders correctly', () => {
    const props = {
        openLayer: jest.fn()
    };
    const component = shallow(<Header {...props} />);

    component.find('.info-button').simulate('click', mockedEvent);

    expect(toJson(component)).toMatchSnapshot();
    expect(props.openLayer).toHaveBeenCalled();
});
