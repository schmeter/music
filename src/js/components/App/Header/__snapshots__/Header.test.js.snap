// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="main-header"
>
  <h1
    className="main-headline"
  >
    <Link
      to="/"
    >
      Musik
    </Link>
  </h1>
  <button
    className="button header-button menu-button"
    onClick={[Function]}
  >
    <Icon
      id="bars"
    />
  </button>
  <button
    className="button header-button info-button"
    onClick={[Function]}
  >
    <Icon
      id="info-circle"
    />
  </button>
</header>
`;
